form/src/main/java/org/dhis2/form/ui/FormView.kt

```diff
@@ -532,15 +532,20 @@ class FormView : Fragment() {
     }
 
     private fun openChooserIntent(uiEvent: RecyclerViewUiEvents.OpenChooserIntent) {
-        if (actionIconsActivate && !uiEvent.value.isNullOrEmpty()) {
+        val currentField = viewModel.queryData.value
+        val currentValue = when (currentField?.id) {
+            uiEvent.uid -> currentField.value
+            else -> uiEvent.value
+        }
+        if (actionIconsActivate && !currentValue.isNullOrEmpty()) {
```

Comment:
I will move this logic to the ViewModel and create a test, maybe here you can do something like val value = viewModel.getUpdatedValue(uiEvent.uid) and in the viewmodel check cache versus stored value. What do you think?

Reply:
Sure, you are rigth